### PR TITLE
[MRG] add `sig collect` command

### PR DESCRIPTION
**Pull request by ctb**
Adds `sourmash sig collect` by integrating [sigs-to-manifest.py](https://github.com/sourmash-bio/database-examples/blob/main/sigs-to-manifest.py) into sourmash.

This is the code that's used to maintain a manifest of wort-genomes signatures, among other things - see https://github.com/sourmash-bio/sourmash/issues/1965.

The main motivation behind this command addition is that it's really enabling for certain use cases, as shown most clearly by the way the documentation simplifies 😃 .

Guide into rendered docs from this PR:
- [sig collect docs](https://sourmash--2036.org.readthedocs.build/en/2036/command-line.html#sourmash-signature-collect-collect-manifests-across-databases)
- ["using manifests to refer to collections"](https://sourmash--2036.org.readthedocs.build/en/2036/command-line.html#using-manifests-to-explicitly-refer-to-collections-of-files)
- [advanced database page on manifests](https://sourmash--2036.org.readthedocs.build/en/2036/databases-advanced.html#manifests)

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review & merge @sourmash-bio/devs!

File: src/sourmash/sig/__main__.py [hunk 1]
```diff
@@ -1384,6 +1396,99 @@ def check(args):
...
+    # convert to abspath
+    if args.abspath:
+        args.locations = [ os.path.abspath(iloc) for iloc in args.locations ]
+
+    # iterate through, loading all the manifests from all the locations.
+    for n_files, loc in enumerate(args.locations):
+        notify(f"Loading signature information from {loc}.")
+
+        if n_files and n_files % 100 == 0:
+            notify(f'... loaded {len(collected_mf)} sigs from {n_files} files')
```
Comment (bluegenes) [inline src/sourmash/sig/__main__.py]:
```suggestion
        if n_files % 100 == 0:
            notify(f'... loaded {len(collected_mf)} sigs from {n_files} files')
```
would this allow it to be covered by tests (report on 0th file)?

File: tests/test_cmd_signature_collect.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+
+    locations = set([ row['internal_location'] for row in manifest.rows ])
+    assert protzip in locations
+    assert hpzip in locations
+    assert dayzip in locations
+    assert len(locations) == 3, locations
+
+
+def test_sig_collect_3_multiple_use_fromfile(runtmp, manifest_db_format):
+    # collect a manifest from two .zip files
```
Comment (bluegenes) [inline tests/test_cmd_signature_collect.py]:
```suggestion
    # collect a manifest from two .zip files using --from-file
```

**Review by bluegenes — APPROVED** [review]
a couple minor suggestions, otherwise lgtm!

**Comment by ctb** [comment]
thank you :)

**Pull request merged**